site/en/docs/extensions/migrating/rhc/index.md

```diff
@@ -0,0 +1,178 @@
+
+
+## What to do if a library is requesting the code
+
+Regardless of where the code comes from, it is not allowed to have RHC. This includes code you didn't author, but just happen to use as a dependency in your project. [Some developers][firebase-rhc-issue] using [Firebase][firebase] had this frustration when remote code was being included for use in [Firebase Auth][firebase-auth]. Even though this was a first party (i.e. Google owned) library - no exception is given for RHC. That code needs to be updated to either remove the RHC, or be able to be configured to not include it in the first place. If you end up hitting an issue where it isn't _your _code that is loading RHC, but a library that you are using, then the best course of action is to contact the library's author. Let them know that this is happening, and ask for either a workaround or code updates to remove it.
+
+
+## What if you can't wait for a library update
+
+Some libraries will ship an update almost immediately after being notified, but others may be abandoned or take time to address the issue. Depending on _what_ is happening in the specific violation, you may not need to wait for them to move to be unblocked and complete a successful review. There are a number of options available to get back up and running quickly.
+
+
+
+### Audit the code
+
+Are you _certain_ that the code that is causing the request is needed? If it can just be deleted, or a library that is causing it can be removed - then we have an easy fix -  delete the code, and the job is done.
+
+Alternatively, is there another library that offers the same functionality? Try checking [npmjs.org](npmjs.org), github, or other sites for other options that fufill the same usecases.
+
+### [Tree Shaking][tree-shaking]
+
+If the code causing the RHC violation isn't actually being used, then it may be able to be automatically deleted by tooling. Modern build tools like [webpack](https://webpack.js.org/), [Rollup](https://rollupjs.org/), and [Vite](https://vitejs.dev/) (just to name a few) have a feature called tree-shaking. Once enabled on your build system, tree shaking should remove any and all unused code paths. This can mean that you not only have a more compliant version of your code, but a leaner and faster one too! It is important to note that not all libraries are able to be tree shaken, but many are. Some tools, like Rollup and Vite, have tree-shaking enabled by default. webpack [needs to be configured](https://webpack.js.org/guides/tree-shaking/) for it to be enabled. If you aren't using a build system as a part of your extension, but _are_ using code libraries, then we really encourage you to investigate adding a build tool to your workflow. They can help you write safer, more reliable and more maintainable projects.
```

Comment:
Nit, since this seems unnecessary and there isn't really a perfect tree-shaking tool :)

```suggestion
If the code causing the RHC violation isn't actually being used, then it may be able to be automatically deleted by tooling. Modern build tools like [webpack](https://webpack.js.org/), [Rollup](https://rollupjs.org/), and [Vite](https://vitejs.dev/) (just to name a few) have a feature called tree-shaking. Once enabled on your build system, tree shaking should remove any unused code paths. This can mean that you not only have a more compliant version of your code, but a leaner and faster one too! It is important to note that not all libraries are able to be tree shaken, but many are. Some tools, like Rollup and Vite, have tree-shaking enabled by default. webpack [needs to be configured](https://webpack.js.org/guides/tree-shaking/) for it to be enabled. If you aren't using a build system as a part of your extension, but _are_ using code libraries, then we really encourage you to investigate adding a build tool to your workflow. They can help you write safer, more reliable and more maintainable projects.
```